source/utils/RandomUtils.cc

```diff
@@ -33,11 +33,14 @@ namespace nexus {
         G4double sinTheta2 = 1. - cosTheta*cosTheta;
         if (sinTheta2 < 0.)  sinTheta2 = 0.;
         G4double sinTheta  = std::sqrt(sinTheta2);
-        G4double phi = CLHEP::twopi*G4UniformRand();
-        if (phi > phi_min && phi < phi_max){
-          return G4ThreeVector (sinTheta*std::cos(phi),
-                                sinTheta*std::sin(phi),
-                                cosTheta).unit();
+        G4bool phi_component = false;
+        while (phi_component == false) {
```

Comment:
Is there any reason why these two lines can't be replaced with `while (true) {` ?

Reply:
... and the same goes for a similar situation in the outer loop and the `dir` flag.

Reply:
Right, done.